contracts/core/Allo.sol

```diff
@@ -1,89 +1,289 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity 0.8.19;
 
-import { Metadata } from "./libraries/Metadata.sol";
+import "@openzeppelin-upgradeable/proxy/ClonesUpgradeable.sol";
 import "@openzeppelin-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgradeable/utils/MulticallUpgradeable.sol";
+import "@openzeppelin-upgradeable/token/ERC20/IERC20Upgradeable.sol";
 
-contract Allo is Initializable {
-    struct PoolData {
-        address identity;
-        address allocationStrategy;
-        address distributionStrategy;
+import "@solady/auth/Ownable.sol";
+
+import {Metadata} from "./libraries/Metadata.sol";
+import "../interfaces/IAllocationStrategy.sol";
+import "../interfaces/IDistributionStrategy.sol";
+import "./Registry.sol";
+
+contract Allo is Initializable, Ownable, MulticallUpgradeable {
+    error NO_ACCESS_TO_ROLE();
+
+    error TRANSFER_FAILED();
+
+    /// @notice Struct to hold details of an Pool
+    struct Pool {
+        bytes32 identityId;
+        IAllocationStrategy allocationStrategy;
+        IDistributionStrategy distributionStrategy;
         Metadata metadata;
+        address token;
+        uint256 amount;
         bool active;
```

Comment:
remove and have strategy implement getter function

Reply:
Latest PR reflects these changes.

Note: @zobront @thelostone-mc @kurtmerbeth how do we want to handle funding the pool?
        1. Do we expose a fundPool() function on the interface?
        2. We still do the transfer to the treasury in the core contract.
        3. Pool transfers happen in the strategies.
       
I know we have ongoing conversation on this here: #5 

@zobront @KurtMerbeth @thelostone-mc WDYT?

Reply:
`fundPool()` in core that takes a fee and sends the funds on to the distribution strategy